clarify implicit C, BBR, RS parameters and relationship of RS and P in stretch ACKs (#83)
ietf-wg-ccwg/draft-ietf-ccwg-bbr: draft-ietf-ccwg-bbr.md

Variables that are not defined below are defined in
{{delivery-rate-samples}}, "Delivery Rate Samples".

In the pseudocode in this document, all functions have implicit access to the
(C) connection state and (BBR) congestion control algorithm state for that
connection. All functions involved in ACK processing additionally have implicit

> access to the the (RS) for per-ack rate sample state for the rate sample
> populated during that ACK processing episode.

Reviewer: ```suggestion
access to the the (RS) for for the rate sample populated processing that ACK.
```

#### Upon receiving an ACK {#upon-receiving-an-ack}

When an ACK arrives, the connection first calls InitRateSample() to initialize
the per-ACK RateSample RS:

~~~~
  /* Initialize the rate sample generated using the ACK being processed. */
  InitRateSample():
    RS.rtt           = -1
    RS.has_data      = false
    RS.prior_time    = 0
    RS.interval      = 0
    RS.delivery_rate = 0
~~~~

> Next, for each packet that was newly acknowledged, the connection calls

Reviewer: ```suggestion
Next, for each newly acknowledged packet, the connection calls
```

UpdateRateSample() to update the per-ACK rate sample based on a snapshot of

> connection delivery information from the time at which the packet was last

Reviewer: Last transmitted is a bit odd for non-TCP transports, since in QUIC a packet is only transmitted once.  Can we just drop 'last' and leave it as an implementation detail?

```suggestion
connection delivery information from the time at which the packet was
```

  /* Is the given Packet the most recently sent packet
   * that has been delivered? */
  IsNewestPacket(Packet P):
    return (P.send_time > C.first_send_time or
            (P.send_time == C.first_send_time and
             after(P.end_seq, RS.last_end_seq))
~~~~

> Finally, after the connection has processed all packets acknowledged by this

Reviewer: ```suggestion
Finally, after the connection has processed all newly acknowledged packets for this
```
